### ISSUE TITLE
Proposed Gitter room restructuring

### ISSUE THREAD

**Issue by QuincyLarson** [freeCodeCamp/freeCodeCamp#15613]
In order to make it easier for people to get started with Gitter - and to increase response time - 

### I propose we merge some of the freeCodeCamp Gitter rooms:

- HelpJavaScript, HelpFrontEnd, HelpBackEnd, and HelpDataViz would be merged into Help. This would eliminate ambiguity and ensure that people who need help end up joining a room where people are around to help them.

- freeCodeCamp would be merged into Casual. 

- CoreTeam is already being merged into Contributors

- CodingJobs would be closed in favor of using https://forum.freecodecamp.com/c/getting-a-developer-job

- CodeReview would be closed in favor of using https://forum.freecodecamp.org/c/project-feedback?max_posts=1

The resulting list of rooms would make LetsPair, and YouCanDoThis, and Contributors more prominent, which should increase the number of people joining those rooms. 

### Thus the resulting directory of rooms found at https://gitter.im/FreeCodeCamp/home would be:
- Help
- Casual
- LetsPair
- YouCanDoThis
- Contributors
- DataScience
- Admin
- Python
- Linux
- PairProgrammingWomen
- Chinese
- Espanol

These are just my thoughts based on using Gitter and talking with highly active Gitter users. 

Before we start the process of backing up all these rooms and consolidating them, I'd like to get feedback from the community, so please feel free to respond with your thoughts below.

**Comment by mstellaluna** [comment]
The main room has been the primary place for casual banter, general discussion. There is a camaderie in this room that can't be replaced. If need be rename the room to General Chat but the main should stay and casual merged into main.

**Comment by breese8009** [comment]
I am not a fan, seems like it would cause more chaos in chatrooms...

**Comment by CodeDraken** [comment]
Campers may see things they don't want to see in a free for all help room.

**Comment by luishendrix92** [comment]
The general chat room has its ups and downs, but it has served as a way to form a community in ways that no other site has done before. I get it that the forum is getting a lot of attention and priority but the general chat has been part of FCC for like 3 years I think.

**Comment by Cyclokitty** [comment]
I love the chaos, discussions, friendships, battles, code advice, life advice (good and terrible), and the cool people I've chatted with over the last nearly two years. Please don't hide the chat in a swirl of rooms.

**Comment by mstellaluna** [comment]
the help rooms should be divided. the general chat should be available and then sub-divide as per the new beta curriculum. having everyone in one general chat will cause a chaos in the inability to track who is speaking to who and who is asking for help.

**Comment by Manish-Giri** [comment]
> HelpJavaScript, HelpFrontEnd, HelpBackEnd, and HelpDataViz would be merged into Help. This would eliminate ambiguity and ensure that people who need help end up joining a room where people are around to help them.

I disagree with this. The separation of rooms has worked out really well thus far. Most of the regulars interested in JS hang out more often in the Help JS Room, as compared to Help. Same goes for folks hanging out in Help Room.

For sake of compactness (if that's what is being achieved here), you could merge HelpFrontEnd, HelpBackEnd, and HelpDataViz into one room, something like "Help FullStack". 

But I feel the Help and HelpJS rooms should be left as is. A person who has just joined FCC today and is looking for help with "Uncommenting HTML" doesn't necessarily need to sift through a wall of posts on how to fetch response from the server or how to render SVG elements correctly. Having 20 people talk about 50 different things at once will just produce clutter, and it will be just plain hard for people talking on a specific topic to communicate with each other.

**Comment by U-ways** [comment]
I really think that the help room should be divided into:
- JavaScript
- HTML&CSS

They both are different interests. And from my experience, some campers would prefer to help with HTML&CSS related problems or JavaScript specific problems. Last of all, both rooms – HelpJavaScript and help – do generate their own "set" of users...

> freeCodeCamp would be merged into Casual.

Regarding the freeCodeCamp & Casual merge, I do think it's a good idea. But I really can't judge whatever Casual should be merged into freeCodeCamp or vice versa. (In my point of view, the room name "freeCodeCamp" has a nicer naming to it – and it does sometimes serve for answers to general questions. But we still got to hear more options about that to see what's best.)

**Comment by QuincyLarson** [comment]
Thanks for your feedback so far everyone. Some of the rooms like HelpDataViz and Casual aren't used very often at all. 

The main challenge with inactive rooms is that campers join them, introduce themselves or ask a question, and don't get a response.

Also, if a camper needs help with a web app, it may not be clear whether a camper should join the "Help", "HelpJavaScript" or "HelpFrontEnd"

My goal in restructuring the rooms is to make this a bit less ambiguous.

Gitter doesn't let you rename rooms, though I can try to contact their team and see whether they can manually rename rooms in their database. If so, would "general" be a better name for the main chat room?

**Comment by mstellaluna** [comment]
General Chat is fine for the main room, then maybe Help can be Help -HTML/CSS like Uway suggested...and helpJavascript can stay the as it.

**Comment by baseballpoetess** [comment]
I think merging Casual > FreeCodeCamp is a good idea so long as it's in that direction.  Trying to push the regulars from fcc into Casual is a baaad idea, IMO.  I'm a relative n00b, only been here since April, so maybe my voice doesn't have as much oomph as some of the others here....but some points to ponder.
- n00bs feel intimidated by advanced questions Segregating HTML/CSS from simple jQuery from more advanced stuff makes *lots* of sense. 
- there is a community in the main FreeCodeCamp room that could be lost if restructuring isn't handled with care. Merging should be done **to** that room rather than from it.
- Room names should be depictive.  FCC-HTMLandCSS for instance.  Keep the branding so we can recruit non-campers who wander in.

If I think of more, I'll add it.

**Comment by Margaret2** [comment]
Hi, just wanted to add my thoughts: 
- Javascript and Back End-related questions sometimes take a long time to help people with and involve pasting a huge amount of code. Having all of this in the general Help room would become overwhelming, especially for someone who just wants to ask a simple question about Bootstrap. 
- On the other hand, people often come to the Help room with Front End related questions, so merging those rooms wouldn't change much. 

So my suggestion is to: 
1. Merge Help Back End and Help Dataviz into Help Javascript.
2. Merge Help Front End into Help.

This would keep related questions together and avoid making either room over-active or chaotic.

**Comment by qmikew1** [comment]
I agree with a lot of the comments - Merging all possible topics _(without separation like js, html/css etc)_ is sort of a recipe for more problems. Just in terms of scrolling-screen real estate you're going to have a mix of unformatted code walls and topics from a-z and the result is
likely going to be pure chaos with an even higher abandonment rate _(for both campers asking questions and those answering)_.

Probably not a popular stance, but why not add some sort of onboarding piece *(that's welcoming)* that would include:

1. What the community is and its purpose
2. Where to ask a question (i.e., maybe examples of different scenarios paired to room)
3. How to ask (and format a question) 

Overall, with respect to community - hands down, it's what separates fcc from a lot of other things 
(warts and all, doesn't mean improvements can't be made
but, the main chat room has historically been a large part of what defines community)

**Comment by Ellyria** [comment]
I'm not a highly active Gitter user, but I do use it, so I'd like to jump in and suggest that there should be **HelpCSS** and **HelpHTML** and **HelpJavaScript**. I realize people will end up in the wrong room sometimes, but users in the rooms they end up in can steer them to rooms that might be more likely to contain people who have an interest in exactly the type of issue they're having or who specialize in that language.

**Comment by Ellyria** [comment]
I thought I'd add that **General** would definitely be a good new name for the **FreeCodeCamp** channel since it doesn't have any negative connotations the way **Casual** can/might/does.

**Comment by jimtryon** [comment]
I also agree that the name **General** is a much more identifiable name than **Casual** for the main FreeCodeCamp channel. I like the idea of merging the rooms together to one room **Help:** HelpJavaScript, HelpFrontEnd, HelpBackEnd, and HelpDataVIs. The only problem I could see with this is that that there will be no separation if a camper is working on a frontend or a backend project.  

The other changes would help to clear up some of the difficulty in finding their way around on FreeCodeCamp.

**Comment by moT01** [comment]
I, personally, like the helpDataViz, helpFrontEnd, and helpBackEnd rooms, ...they line up with curriculum. The frontEnd stays pretty busy - the other two come in bursts, but it's nice to have rooms that aren't so busy as well, and a place for specific questions/topics. I just started the backend section and I have used each of the rooms as I moved through the projects, and they have been useful to me.

**Comment by nsuchy** [comment]
The codingjobs chat room has a spam problem so I agree with closing it.

**Comment by QuincyLarson** [comment]
Thank you for your feedback everyone. My general impression from reading through everyone's feedback is that we should leave the chatrooms as they are. I will close this and we may revisit this at some point in the future. Happy coding!

**Comment by QuincyLarson** [comment]
Several campers have pointed out that coding-jobs is mostly spammers coming in to try and trick campers with dubious job offers. So the one action I'm going to take is to close that room. If anyone is interested in talking about getting a developer job, [we have a popular forum category for that](https://forum.freecodecamp.com/c/getting-a-developer-job?order=views), and it's easier to moderate.

**Comment by mstellaluna** [comment]
With beta on the way, I think we should re-open this discussion. The chat rooms (and their description) should better match the beta map layout. 
example:
FreeCodeCamp main room -- main can handle any requests/general questions, welcomes, account issues
Help -> Merge into the FreeCodeCamp main room
HelpJavascript -> Can handle the JS section in beta 
YouCanDoThis -> can be merged into casual ... there is already has a forum section for this ... 
CodeReview -> Can be requested in any room or the room appropriate for the section
HelpFrontEnd -> FrontEnd section of beta
HelpBackEnd -> React, Node, MongoDB,Express

The rooms that are not mentioned can stay as they are perhaps give a more clearer description of the room's purpose and provide a link to the room list if they are in the wrong room. 

Also a modal prior to entering the room with a link to our code of conduct, room description would be a great addition.
@QuincyLarson
